tests/sqlparser_postgres.rs

```diff
@@ -1137,7 +1137,7 @@ fn parse_array_index_expr() {
         .collect();
 
     let sql = "SELECT foo[0] FROM foos";
-    let select = pg().verified_only_select(sql);
+    let select = pg_and_generic().verified_only_select(sql);
```

Comment:
>Just waiting on a test and this one is good to go

@alamb Ops, I missed your message. I added generic dialect to the `parse_array_index_expr` test to prove that it works with generic dialect too.